src/Interpreters/Session.cpp

```diff
@@ -439,11 +439,12 @@ ContextMutablePtr Session::makeQueryContextImpl(const ClientInfo * client_info_t
     ContextMutablePtr query_context = Context::createCopy(from_session_context ? session_context : global_context);
     query_context->makeQueryContext();
 
+    auto query_context_user = query_context->getUser();
     LOG_DEBUG(log, "{} Creating query context from {} context, user_id: {}, parent context user: {}",
               toString(auth_id),
               from_session_context ? "session" : "global",
               user_id ? toString(*user_id) : "<EMPTY>",
-              query_context->getUser() ? query_context->getUser()->getName() : "<NOT SET>");
+              query_context_user ? query_context_user->getName() : "<NOT SET>");
```

Comment:
It is ok that we can execute a query without any user? Maybe it's ok for some internal queries, but seem like it's not ok when `from_session_context` is true

Reply:
(see https://github.com/ClickHouse/ClickHouse/pull/35763#discussion_r838926531)

Reply:
Later on we check that user is non-null, and properly set it for produced session, etc. But at this particular spot we are trying to prevent a crash.

And what is even more interesting: it can happen even if there WAS a user when `ContextAccess` was generated, but that user could be `DROP`ped from another query, leading to `nullptr` as user in `ContextAccess`.

Reply:
I understand that it may happen when user was dropped concurrently. I mean that messages like `Creating query context from session context, user_id: <NOT SET>` in logs look creepy. We should avoid such messages.